File: CS/dxKB13085/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrinting.Control;
using DevExpress.XtraReports.UserDesigner;

namespace dxKB13085 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        XRDesignFormEx designForm;

        private void button1_Click(object sender, System.EventArgs e) {
            designForm = new XRDesignFormEx();
            designForm.DesignPanel.DesignerHostLoaded += new DesignerLoadedEventHandler(OnDesignerLoaded);
            designForm.DesignPanel.ExecCommand(ReportCommand.NewReport);
            designForm.Show();
        }

        private void OnDesignerLoaded(object sender, DesignerLoadedEventArgs e) {
            designForm.DesignPanel.Report.Scripts.OnAfterPrint = "private void OnAfterPrint(object sender, " +
                "System.EventArgs e) {((XtraReport)sender).PrintingSystem.AddCommandHandler(new " +
                "dxKB13085.ZoomInCommandHandler());}";
        }

    }

    public class ZoomInCommandHandler : DevExpress.XtraPrinting.Native.ICommandHandler {
        public virtual void HandleCommand(PrintingSystemCommand command, object[] args, PrintControl control, ref bool handled) {
            if (!CanHandleCommand(command, control)) return;
            MessageBox.Show("Handled!");
            handled = true;
        }

        public virtual bool CanHandleCommand(PrintingSystemCommand command, PrintControl control) {
            return command == PrintingSystemCommand.ZoomIn;
        }
    }
}

File: VB/dxKB13085/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrinting.Control
Imports DevExpress.XtraReports.UserDesigner

Namespace dxKB13085
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private designForm As XRDesignFormEx

        Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button1.Click
            designForm = New XRDesignFormEx()
            AddHandler designForm.DesignPanel.DesignerHostLoaded, AddressOf OnDesignerLoaded
            designForm.DesignPanel.ExecCommand(ReportCommand.NewReport)
            designForm.Show()
        End Sub

        Private Sub OnDesignerLoaded(ByVal sender As Object, ByVal e As DesignerLoadedEventArgs)
            designForm.DesignPanel.Report.Scripts.OnAfterPrint = "private void OnAfterPrint(object sender, " _
            & "System.EventArgs e) {((XtraReport)sender).PrintingSystem.AddCommandHandler(new " _
            & "dxKB13085.ZoomInCommandHandler());}"
        End Sub

    End Class

    Public Class ZoomInCommandHandler
        Implements DevExpress.XtraPrinting.Native.ICommandHandler

        Public Overridable Sub HandleCommand(ByVal command As PrintingSystemCommand, ByVal args As Object(), ByVal control As PrintControl, ByRef handled As Boolean) _
        Implements DevExpress.XtraPrinting.Native.ICommandHandler.HandleCommand
            If (Not CanHandleCommand(command, control)) Then
                Return
            End If
            MessageBox.Show("Handled!")
            handled = True
        End Sub

        Public Overridable Function CanHandleCommand(ByVal command As PrintingSystemCommand, ByVal control As PrintControl) As Boolean _
            Implements DevExpress.XtraPrinting.Native.ICommandHandler.CanHandleCommand

            Return command = PrintingSystemCommand.ZoomIn
        End Function
    End Class
End Namespace